### PR TITLE
Fix/get credentials for multiple GitHub apps

### PR DESCRIPTION
**Pull request by chicoribas**
## Hey, I just made a Pull Request!

When multiples Github Apps was configured for integrations, only the first credential was returned due a mistake on find logic.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/sweet-chairs-draw.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-catalog-backend-module-github': patch
+---
+
+Bump integration package
```
Comment (freben) [inline .changeset/sweet-chairs-draw.md:5]: And this message updated (explain to the reader what changed)

**Review by freben — COMMENTED** [review]
Looks good, apart from a bit of changeset fiddling.

File: .changeset/sweet-chairs-draw.md (hunk 1 above)
Comment (chicoribas) [reply .changeset/sweet-chairs-draw.md:5]: Maybe I missed something: the package @backstage/plugin-catalog-backend-module-github shouldn't be deployed to bump @backstage/integration version?

**Comment by freben** [comment]
You still need to delete `.changeset/sweet-chairs-draw.md`.

**Comment by chicoribas** [comment]
> You still need to delete `.changeset/sweet-chairs-draw.md`.

File deleted

Thanks...

File: .changeset/good-lions-approve.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/integration': patch
+---
+
+Bug fix for multiples github apps integration
```
Comment (Rugvip) [inline .changeset/good-lions-approve.md:5]:
```suggestion
Fixed a bug where the wrong credentials would be selected when using multiple GitHub app integrations.
```

**Review by Rugvip — APPROVED** [review]
Thank you! 👍 

Looks good, just tweaking the changeset a bit

**Pull request merged**